src/Microsoft.Health.Fhir.Core/Features/Search/SearchOptions.cs

```diff
@@ -97,6 +97,16 @@ internal set
         /// </summary>
         public IReadOnlyList<(SearchParameterInfo searchParameterInfo, SortOrder sortOrder)> Sort { get; internal set; }
 
+        /// <summary>
+        /// Marks whether we need to execute the second set of queries for (certain types of) sort.
+        /// </summary>
+        public bool SortQuerySecondPhase { get; internal set; } = false;
+
+        /// <summary>
+        /// Sets whether this search query is of type sort with filter.
+        /// </summary>
+        public bool IsSortWithFilter { get; internal set; } = false;
+
```

Comment:
I would argue what both of this properties is SQL specific and shouldn't be part of this object. 

As option we can have SqlSearchOptions derived form SearchOptions and use it inside sql library only

Reply:
Updated code to have new SqlSearchOptions class.